lib/gollum/app.rb

```diff
@@ -465,7 +465,12 @@ class App < Sinatra::Base
         if @versions.size == 1
           wikip  = wiki_page(params[:splat].first)
           commit = wikip.wiki.repo.commit(@versions.first)
-          @versions.push(commit.parent.id)
+          parent = commit.parent
+          if parent.nil?
+            redirect to("#{@file}/#{@commit.id}")
```

Comment:
Just to make sure -- this is the method that takes account of the base path, right? If so, I think this is ready to merge!

Reply:
Yes, it redirects to `/base_path/path/to/page/sha1`.